underscore.js

```diff
@@ -1009,11 +1009,21 @@
 
   // Convert an object into a list of `[key, value]` pairs.
   _.pairs = function(obj) {
-    var keys = _.keys(obj);
-    var length = keys.length;
-    var pairs = Array(length);
-    for (var i = 0; i < length; i++) {
-      pairs[i] = [keys[i], obj[keys[i]]];
+    var pairs, i, length;
+    if (_.isMap(obj)) {
+      var entries = obj.entries();
```

Comment:
Yeah, I agree that `Array.from` is a cleaner solution. Unfortunately it isn't supported in node 0.12.*, hence the failed tests.  
```javascript
> typeof Map
'function'
> Array.from
undefined
```
Not sure on the question of whether or not to even use underscore with Maps. I saw [#2517](https://github.com/jashkenas/underscore/issues/2517#issuecomment-216385065) and decided to submit a PR.